### ISSUE TITLE
justenoughdimensions startup crash

### ISSUE THREAD

**Issue by Bennyboy1695** [SpongePowered/SpongeForge#2150]
**I am currently running**

- SpongeForge version: spongeforge-1.12.2-2655-7.1.0-BETA-3074
- Forge version: forge-1.12.2-14.23.3.2655-universal
- Java version: 1.8
- Operating System: Ubuntu 16.04

- Plugins/Mods: Modpack Exoria, No plugins yet as server was in crash loop

**Issue Description**
Server is in a constant crash loop because its unable to load the worlds due to error mentioned in crash report. Pack works fine without sponge but the error appears as soon as sponge is installed.

Crash report - https://gist.github.com/Bennyboy1695/93c3bd0adce474aea88c31633077f637
Crash report Remapped - https://gist.github.com/ryantheleach/bfa6c2529fe3fa4ec5813d483979176a

Log - https://gist.github.com/Bennyboy1695/9f738bf2a588115527182a12d43aa311

**Comment by phit** [comment]
this caused by justenoughdimensions and its config in this pack

**Comment by ryantheleach** [comment]
https://github.com/maruohon/justenoughdimensions/issues/9

@maruohon Seems relatively Sponge friendly on this issue, Maybe we could work something out?


This is the config that ships with Exoria-Server-1.3.0c.zip which contains: justenoughdimensions-1.12.2-1.6.0-dev.20180502.030835.jar

Which is the latest version of the modpack at time of posting.

I've not tested this.

https://gist.github.com/ryantheleach/b4e1d985457192d91da3aed0a0c51155

**Comment by maruohon** [comment]
I suspect it's caused by unregistering the Nether and the End in Exoria's config. From trying to look through the SpongeCommon `WorldManager` code on Github, looks like the `dimensionType` variable in the `WorldManager.loadAllWorlds(WorldManager.java:683)` method is null on that line, but I couldn't see why that would happen. Looks like SpongeCommon always adds/registers the vanilla dimensions in its own lists/maps, but I couldn't see why the `DimensionType` would then end up being null when iterating over the map. Would need to have a proper Sponge dev env and look at the call hierarchies to see how stuff actually works... but I don't have that, plus I'm already quite swamped with just Forge and Liteloader side mod stuff. So maybe some Sponge devs can shed some light on how this stuff works in Sponge?

Basically on JED's side, I allow unregistering dimensions based on the config. The unregistering is just this Forge `DimensionManager` method call: https://github.com/maruohon/justenoughdimensions/blob/master/src/main/java/fi/dy/masa/justenoughdimensions/config/DimensionConfig.java#L308
and it happens from here: https://github.com/maruohon/justenoughdimensions/blob/master/src/main/java/fi/dy/masa/justenoughdimensions/JustEnoughDimensions.java#L81
